### PR TITLE
feat: move smtp and cors from static to dynamic configuration

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes https://github.com/gravitee-io/issues/issues/4432
Fixes https://github.com/gravitee-io/issues/issues/4433

### CODE REVIEW

File: src/management/configuration/portal/portal.component.ts [hunk 1]
```diff
@@ -72,6 +87,26 @@ const PortalSettingsComponent: ng.IComponentOptions = {
...
+      return query ? this.headers.filter(this.createFilterFor(query)) : [];
+    };
+
+    this.createFilterFor = (query: string) => {
+      let lowercaseQuery = query.toLowerCase();
+
+      return function filterFn(header) {
+        return header.toLowerCase().indexOf(lowercaseQuery) === 0;
+      };
+    };
```
Comment (phiz71) [inline src/management/configuration/portal/portal.component.ts:109]: Add this into `cors.services.ts`

File: src/services/cors.service.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class CorsService {
+  private Constants: any;
+
+  private allowOriginPattern = '^(?:(?:[htps\\(\\)?\\|]+):\\/\\/)*(?:[\\w\\(\\)\\[\\]\\{\\}?\\|.*-](?:(?:[?+*]|\\{\\d+(?:,\\d*)?\\}))?)+(?:[a-zA-Z0-9]{2,6})?(?::\\d{1,5})?$';
```
Comment (phiz71) [inline src/services/cors.service.ts:19]: Remove this field from ApiProxyController

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: src/management/configuration/portal/portal.component.ts (hunk 1 above)
Comment (ytvnr) [reply src/management/configuration/portal/portal.component.ts:109]: done

File: src/services/cors.service.ts (hunk 2 above)
Comment (ytvnr) [reply src/services/cors.service.ts:19]: resolved

**Review by phiz71 — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]

**Pull request merged**